internal/messagejournal/messagejournal.go

```diff
@@ -0,0 +1,231 @@
+	if err != nil {
+		return nil, fmt.Errorf("cannot build dynamic sql query: %w", err)
+	}
+
+	preparedQuery, err := j.database.Prepare(queryString)
+	if err != nil {
+		return nil, fmt.Errorf("cannot prepare query when retrieving journal entries: %w", err)
+	}
+
+	rows, err := preparedQuery.Query()
+	if err != nil {
+		return nil, fmt.Errorf("cannot execute query to retrieve journal entries: %w", err)
+	}
+
+	for rows.Next() {
+		var rowID int
+		var messageID string
+		var sent time.Time
+		var workerName string
+		var responseTo string
+		var workerEvent uint
+		var workerEventMessage string
+
+		err := rows.Scan(
+			&rowID,
+			&messageID,
+			&sent,
+			&workerName,
+			&responseTo,
+			&workerEvent,
+			&workerEventMessage,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("cannot scan journal entry columns: %w", err)
+		}
+
+		// Truncate the worker messages by the truncate length specified.
+		messageMaxSize := len(workerEventMessage)
+		if messageMaxSize >= filter.TruncateLength {
+			messageMaxSize = filter.TruncateLength
+			workerEventMessage = fmt.Sprintf("%+v...", workerEventMessage[:messageMaxSize])
+		}
+
+		// Convert the entry properties into a string format and append to the list of entries.
+		newMessage := map[string]string{
+			"message_id":     messageID,
+			"sent":           sent.String(),
+			"worker_name":    workerName,
+			"response_to":    responseTo,
+			"worker_event":   ipc.WorkerEventName(workerEvent).String(),
+			"worker_message": workerEventMessage,
+		}
+		entries = append(entries, newMessage)
+	}
+	rowIterationErr := rows.Err()
+	if rowIterationErr != nil {
+		return nil, fmt.Errorf("cannot iterate queried journal entries: %w", rowIterationErr)
+	}
+	closeErr := rows.Close()
+	if closeErr != nil {
+		return nil, fmt.Errorf("cannot close journal entry rows: %w", closeErr)
+	}
+
+	return entries, nil
+}
+
+// buildDynamicGetEntriesQuery is a utility method that builds the dynamic sql query
+// required to filter journal entry messages from the message journal database
+// when they are retrieved in the 'GetEntries' method.
+func (j *MessageJournal) buildDynamicGetEntriesQuery(filter Filter) (string, error) {
+	queryTemplate := template.New("dynamicGetEntriesQuery")
+	queryTemplateParse, err := queryTemplate.Parse(
+		`SELECT * FROM {{.Table}}
+		{{if .MessageID}} INTERSECT SELECT * FROM {{.Table}} WHERE message_id='{{.MessageID}}'{{end}}
+		{{if .Worker}} INTERSECT SELECT * FROM {{.Table}} WHERE worker_name='{{.Worker}}'{{end}}
+		{{if .From}} INTERSECT SELECT * FROM {{.Table}} WHERE sent>='{{.From}}'{{end}}
+		{{if .To}} INTERSECT SELECT * FROM {{.Table}} WHERE sent<='{{.To}}'{{end}}
+		{{if not .Persistent}} INTERSECT SELECT * FROM {{.Table}} WHERE sent>='{{.InitializedAt}}'{{end}}
+		ORDER BY sent`,
+	)
+	if err != nil {
+		return "", fmt.Errorf("cannot parse query template parameters: %w", err)
+	}
+	var compiledQuery bytes.Buffer
+	queryCompileErr := queryTemplateParse.Execute(&compiledQuery,
```

Comment:
You don't need to allocate a new error variable here. You've declared `err`
already, and by the time code reaches this line, you can just assign the return
value of `Execute` to `err`.


```suggestion
	err = queryTemplateParse.Execute(&compiledQuery,
```